src/lib/hooks/use-logout-user.ts

```diff
@@ -0,0 +1,53 @@
+import { useUserSessionMutation } from "@/lib/api/user-sessions";
+import { useAuthStore } from "@/lib/providers/auth-store-provider";
+import { useCoachingRelationshipStateStore } from "@/lib/providers/coaching-relationship-state-store-provider";
+import { useCoachingSessionStateStore } from "@/lib/providers/coaching-session-state-store-provider";
+import { useOrganizationStateStore } from "@/lib/providers/organization-state-store-provider";
+import { useRouter } from "next/navigation";
+
+export function useLogoutUser() {
```

Comment:
I used a hook pattern here as that's what seemed to make sense. Not sure if that's idiomatic for this sort of thing or if it would be better served as just some sort of utility function. Would love feedback on this.

Reply:
I think it works great, I love the addition of this hook and it doesn't seem unnatural at all.